lib/compress.js

```diff
@@ -3112,6 +3056,63 @@ merge(Compressor.prototype, {
                 }
             }
         }
+        if (compressor.option("unsafe_Func")
+            && exp instanceof AST_SymbolRef
+            && exp.undeclared()
+            && exp.name == "Function") {
+            // new Function() => function(){}
+            if (self.args.length == 0) return make_node(AST_Function, self, {
+                argnames: [],
+                body: []
+            });
+            if (all(self.args, function(x) {
+                return x instanceof AST_String;
+            })) {
+                // quite a corner-case, but we can handle it:
+                //   https://github.com/mishoo/UglifyJS2/issues/203
+                // if the code argument is a constant, then we can minify it.
+                try {
+                    var code = "NaN(function(" + self.args.slice(0, -1).map(function(arg) {
```

Comment:
Most of the complex use cases won't be affected by `side_effects`, which I assume would be more common than these short examples.

Even the test case in this PR is immune to `side_effects`:
```js
$ echo 'Function("require", "module", "exports", "module.exports = 42;");' | bin/uglifyjs -c unsafe_Func
Function("a","b","c","b.exports=42");
```